src/btree/bt_split.c

```diff
@@ -1061,9 +1069,18 @@ __split_internal(WT_SESSION_IMPL *session, WT_PAGE *parent, WT_PAGE *page)
 	__split_ref_prepare(session, alloc_index,
 	    __wt_session_gen(session, WT_GEN_SPLIT), true);
 
+	/* Encourage a race */
+	__page_split_timing_stress(session,
+	    WT_TIMING_STRESS_INTERNAL_PAGE_SPLIT_RACE, 100 * WT_THOUSAND);
+
 	/* Split into the parent. */
 	if ((ret = __split_parent(session, page_ref, alloc_index->index,
 	    alloc_index->entries, parent_incr, false, false)) == 0) {
+		/* Encourage a race */
+		__page_split_timing_stress(session,
+		    WT_TIMING_STRESS_INTERNAL_PAGE_SPLIT_RACE,
+		    100 * WT_THOUSAND);
+
```

Comment:
I'd suggest moving this stress test into the `__split_parent()` function, that way we'll test other parent split code as well as this path (for example, `__wt_split_reverse()` and `__split_insert()` call `__split_parent()` directly.